src/platform/Linux/DnssdImpl.cpp

```diff
@@ -328,6 +328,7 @@ CHIP_ERROR MdnsAvahi::Init(DnssdAsyncReturnCallback initCallback, DnssdAsyncRetu
     CHIP_ERROR error = CHIP_NO_ERROR;
     int avahiError   = 0;
 
+    Shutdown();
```

Comment:
The API docs say double-init is supported and after the first init should be no-op.

Won't calling Shutdown kill existing resolves?  Also, the double-init should be getting properly detected at a higher level (in the platform mdns code), not in this backend....

Reply:
Making double Init a no-op would be dangerous and even crash if any of the 3 arguments change. If we find a higher layer fix that would be even better, this would still work though and not interfere with it. Regarding killing existing resolves: double init is already an indication that they did not work anyway, also went through whole TE9 with this change without any mdns issues.

Reply:
The API on the higher level is supposed to guarantee that double init is a no-op.  If it's not doing that, we need to fix it.  I agree that we could in theory take this change anyway, and if the higher level is fixed this would just become dead code, I guess....

Reply:
That's my theory too. For now we could go with the symptom fix.

Reply:
Is there something tracking the issue in the higher level API?

Reply:
Added an issue but it is well known since TE9: https://github.com/project-chip/connectedhomeip/issues/18745

Reply:
Known is good, but "known but not tracked" is terrible.

Reply:
Agreed, it happened because it was so easy to add the Shutdown call and forget about it. Right now it's better late than never.